### PR TITLE
Features/update session file

### PR DESCRIPTION
**Pull request by SamirHassoun**
1) Updated Swagger definition for DataServices
2) Code tidy within DataServices
3) Added config for portal regarding SDRA writeable folder
4) Updated portal to use list of usages from UI popup
5) Updated SessionFile to use selected Usages
6) Fixed current unit tests
7) Added new unit tests

### CODE REVIEW

File: src/DataServices/DataServices/Controllers/DataAccessApi.cs [hunk 1]
```diff
@@ -81,7 +81,7 @@ public async Task<IActionResult> DeleteDocumentRequestJobFromQueue([FromRoute][R
                 returnCode = new ReturnCode
                 {
                     Code = task.ErrorCode,
-                    Message = task.Message
+                    Message = task.Message  // contains filename: "\\\\engineering.ukho.gov.uk\\dfs\\AppData\\HDB\\DEV1\\SDRADocuments\\RSDRA2018000065913_1.pdf"
```
Comment (Rossall82) [inline src/DataServices/DataServices/Controllers/DataAccessApi.cs:84]: Can we remove the path here - happy to leave in a comment advising that the filename with path is returned though. Just don't want to have paths in our code. Thanks!

File: src/Portal/Portal.UnitTests/SessionFileGeneratorTests.cs [hunk 2]
```diff
@@ -108,6 +117,39 @@ public async Task Test_PopulateSessionFile_Returns_Valid_Session_File_Contents()
...
+                ProcessId,
+                UserFullName,
+                "Assess",
+                new CarisProjectDetails(), selectedUsages, null);
+
+            Assert.IsNotNull(sessionFile);
+            Assert.IsNotNull(sessionFile.DataSources);
+            Assert.AreEqual(selectedUsages[0],
+                sessionFile.DataSources.DataSource.SourceParam.USAGE);
+            Assert.That(selectedUsages, Is.EqualTo(sessionFile.DataSources.DataSource.SourceParam.SELECTEDPROJECTUSAGES.Value));
```
Comment (dynamictulip) [inline src/Portal/Portal.UnitTests/SessionFileGeneratorTests.cs:151]: Good to see usage of the constraints model however the "actual" value should be on the left and the "expected" value should be the one within the `EqualTo` constraint

File: src/Portal/Portal/Helpers/SessionFileGenerator.cs [hunk 3]
```diff
@@ -112,7 +110,7 @@ public async Task<SessionFile> PopulateSessionFile(int processId, string userFul
                             {
                                 Visible = "true",
                                 Expanded = "false",
-                                Name = "registration test2:Nav 15"
+                                Name = $":HPD:Project:|{carisProjectDetails.ProjectName}:{selectedHpdUsages.First()}"
```
Comment (dynamictulip) [inline src/Portal/Portal/Helpers/SessionFileGenerator.cs:113]: There should be a unit test for this change

File: src/DataServices/DataServices/Controllers/DataAccessApi.cs (hunk 1 above)
Comment (SamirHassoun) [reply src/DataServices/DataServices/Controllers/DataAccessApi.cs:84]: I've added on purpose for the next story just as a reminder

File: src/Portal/Portal.UnitTests/SessionFileGeneratorTests.cs (hunk 2 above)
Comment (SamirHassoun) [reply src/Portal/Portal.UnitTests/SessionFileGeneratorTests.cs]: Fixed, Thanks Sara;

File: src/Portal/Portal/Helpers/SessionFileGenerator.cs (hunk 3 above)
Comment (SamirHassoun) [reply src/Portal/Portal/Helpers/SessionFileGenerator.cs:113]: Fixed, thanks Sara

Comment (dynamictulip) [reply src/Portal/Portal/Helpers/SessionFileGenerator.cs:113]: I see an extended test for `DataSource.SourceString` (good stuff) but I'd still like to see one for `DisplayLayer.Name`

Comment (SamirHassoun) [reply src/Portal/Portal/Helpers/SessionFileGenerator.cs:113]: fixed, thanks

**Pull request merged**